packages/entity-server/src/types.ts

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable @typescript-eslint/no-empty-interface */
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { ModelRegistry } from '@tupaia/database';
+import { EntityModel } from './entity';
+
+export { EntityModel, EntityFields } from './entity';
+export { FetchEntityRequest, FetchEntityResponse, EntityResponseObject } from './routes';
+
+export interface EntityServerModelRegistry extends ModelRegistry {
+  readonly entity: EntityModel;
+}
+
+export type Context<T> = {
+  [field in keyof T]: T[field];
```

Comment:
The result of this looks like a type that is just the same as `T`?

Reply:
You have indeed, I'm worried I'm doing something a bit silly here so if you can see a better way please let me know.

Ideally what we'd be doing here is making context's type itself be a generic parameter of the Request type, ie:
`Request<P = ParamsDict, ..., Context = any>` 

However, you can't extend the number of generic type parameters of an existing type when using declaration merging (which is what you're meant to do to add fields to the Express.Request type).

My solution was to use generics with the `Context` type so that it basically becomes whatever you pass into it, a bit funky admittedly but seems to get the job done.